enterprise/frontend/src/embedding/data-picker/DataSelector/tests/DataSourceSelector.unit.spec.tsx

```diff
@@ -0,0 +1,224 @@
+            tables: [...createNNumberOfTables(10, DB_ID)],
+          }),
+        ],
+      });
+      expect(await screen.findByText("Sample Database")).toBeInTheDocument();
+      expect(screen.getByText("Many tables Database")).toBeInTheDocument();
+
+      await userEvent.click(await screen.findByText("Many tables Database"));
+      expect(await screen.findByText("Table 1")).toBeInTheDocument();
+      expect(await screen.findByText("Table 2")).toBeInTheDocument();
+      expect(await screen.findByText("Table 3")).toBeInTheDocument();
+      expect(await screen.findByText("Table 4")).toBeInTheDocument();
+      expect(await screen.findByText("Table 5")).toBeInTheDocument();
+      expect(await screen.findByText("Table 6")).toBeInTheDocument();
+      expect(await screen.findByText("Table 7")).toBeInTheDocument();
+      expect(await screen.findByText("Table 8")).toBeInTheDocument();
+      expect(await screen.findByText("Table 9")).toBeInTheDocument();
+      expect(await screen.findByText("Table 10")).toBeInTheDocument();
+
+      const searchBox = screen.getByPlaceholderText("Find...");
+      expect(searchBox).toBeInTheDocument();
+
+      // Assert that the search query is used.
+      await userEvent.type(searchBox, "Table 1");
+      expect(await screen.findByText("Table 1")).toBeInTheDocument();
+      expect(screen.queryByText("Table 2")).not.toBeInTheDocument();
+      expect(screen.queryByText("Table 3")).not.toBeInTheDocument();
+      expect(screen.queryByText("Table 4")).not.toBeInTheDocument();
+      expect(screen.queryByText("Table 5")).not.toBeInTheDocument();
+      expect(screen.queryByText("Table 6")).not.toBeInTheDocument();
+      expect(screen.queryByText("Table 7")).not.toBeInTheDocument();
+      expect(screen.queryByText("Table 8")).not.toBeInTheDocument();
+      expect(screen.queryByText("Table 9")).not.toBeInTheDocument();
+      expect(await screen.findByText("Table 10")).toBeInTheDocument();
+    });
+
+    it("should not show search input when there less than 10 tables", async () => {
+      setup(setupOpts);
+      expect(await screen.findByText("Orders")).toBeInTheDocument();
+      expect(screen.getByText("Sample Database")).toBeInTheDocument();
+
+      expect(screen.queryByPlaceholderText("Find...")).not.toBeInTheDocument();
+    });
+  });
+
+  describe("both models and tables are available", () => {
+    const setupOpts: SetupOpts = {
+      availableModels: "tables-and-models",
+    };
+
+    it("should close the picker when clicking outside", async () => {
+      setup(setupOpts);
+
+      expect(await screen.findByText("Models")).toBeInTheDocument();
+      expect(screen.getByText("Raw Data")).toBeInTheDocument();
+
+      await userEvent.click(
+        screen.getByText("Click me to open or close data picker"),
+      );
+
+      expect(screen.queryByText("Models")).not.toBeInTheDocument();
+      expect(screen.queryByText("Raw Data")).not.toBeInTheDocument();
+    });
```

Comment:
I would move this out of the `describe("both models and tables are available"` block since the ability to close the picker doesn't really depend on what entities are available.

Reply:
well you have a point. Fixed in cd3ff1223b81e7d0350ff02cb7edc29d111e9217